### PR TITLE
Add force Outputs to `ExpressionBasedCoordinateForce` and `ExpressionBasedPointToPointForce`

### PR DESCRIPTION
**Pull request by nickbianco**
### Brief summary of changes

Adds `Output`s to access forces computed in `ExpressionBasedCoordinateForce` and `ExpressionBasedPointToPointForce`. 

(Apologies for all of the whitespace trimming changes from my IDE.)

### Testing I've completed

Added tests to `testForces.cpp`.

### CHANGELOG.md (choose one)

- updated.

- - -
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3872)

### CODE REVIEW

**Comment by nickbianco** [comment]
FYI @stingjp @mattpetrucci

File: OpenSim/Simulation/Model/ExpressionBasedCoordinateForce.cpp [hunk 1]
```diff
@@ -53,8 +53,8 @@ ExpressionBasedCoordinateForce::ExpressionBasedCoordinateForce(
     setExpression(expression);
 }
 
-// Set the expression for the force function and create it's lepton program 
-void ExpressionBasedCoordinateForce::setExpression(const string& expression) 
+// Set the expression for the force function and create it's lepton program
```
Comment (tkuchida) [inline OpenSim/Simulation/Model/ExpressionBasedCoordinateForce.cpp:56]: it's -> its

File: OpenSim/Simulation/Test/testForces.cpp [hunk 2]
```diff
@@ -33,6 +33,7 @@
 //      7. RotationalCoordinateLimitForce
 //      8. ExternalForce
 //      9. PathSpring
+//     10. ExpressionBasedCoordinateForce
```
Comment (tkuchida) [inline OpenSim/Simulation/Test/testForces.cpp:36]: 9, 10, 10, 11 -> 9, 10, 11, 12

**Review by tkuchida — COMMENTED** [review]
Nice 😎 Just curious: why not include `ExpressionBasedBushingForce` ?

**Comment by nickbianco** [comment]
> Nice 😎 Just curious: why not include ExpressionBasedBushingForce ?

Other than laziness? No 😅

Added an `Output` for `ExpressionBasedBushingForce` now, which needed some additional changes.

**Review by AllisonJohn — CHANGES_REQUESTED** [review]
I just saw the comment in `ExpressionBasedBushingForce`, everything else seems fine

> Reviewed 1 of 6 files at r1, 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3872#-:-O4NDWkNB3wnad0POOmY:b7dtrnr)* status: all files reviewed, 3 unresolved discussions (waiting on @nickbianco)

___
*[`OpenSim/Simulation/Model/ExpressionBasedBushingForce.h` line 97 at r2](https://reviewable.io/reviews/opensim-org/opensim-core/3872#-O4NAtRH4l_dqT2lGYei:-O4NAtRIBKlnEHuAXxCp:bi3lym4) ([raw file](https://github.com/opensim-org/opensim-core/blob/83b890b6c5e7a0269087ddac9879bede46da5794/OpenSim/Simulation/Model/ExpressionBasedBushingForce.h#L97)):*
> ```objc
> 
> //==============================================================================
> // PROPERTIES
> ```

Should this say "OUTPUTS"?

**Review by nickbianco — COMMENTED** [review]
I realized I was using the incorrect stage for the `Output`s based on how the cache variables work, so I've made some additional improvements.

> *[Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3872#-:-O4NMZI04zBUtXIgvqJ3:b-vt4mdz)* status: 4 of 8 files reviewed, 3 unresolved discussions (waiting on @AllisonJohn and @tkuchida)

___
*[`OpenSim/Simulation/Model/ExpressionBasedBushingForce.h` line 97 at r2](https://reviewable.io/reviews/opensim-org/opensim-core/3872#-O4NAtRH4l_dqT2lGYei:-O4NMUdD3foeN6p32_HC:b4irkar) ([raw file](https://github.com/opensim-org/opensim-core/blob/83b890b6c5e7a0269087ddac9879bede46da5794/OpenSim/Simulation/Model/ExpressionBasedBushingForce.h#L97)):*
<details><summary><i>Previously, AllisonJohn wrote…</i></summary><blockquote>

Should this say "OUTPUTS"?
</blockquote></details>

Yes it should! Fixed.
___
*[`OpenSim/Simulation/Model/ExpressionBasedCoordinateForce.cpp` line 56 at r1](https://reviewable.io/reviews/opensim-org/opensim-core/3872#gh-1717682790:-O4NMXHO5cF98y1EfcPw:b-896fix) ([raw file](https://github.com/opensim-org/opensim-core/blob/418f4d86feff23626383bbe02a9aaa7cd5357290/OpenSim/Simulation/Model/ExpressionBasedCoordinateForce.cpp#L56)):*
<details><summary><i>Previously, tkuchida (Tom Uchida) wrote…</i></summary><blockquote>

it's -> its
</blockquote></details>

Done.
___
*[`OpenSim/Simulation/Test/testForces.cpp` line 36 at r1](https://reviewable.io/reviews/opensim-org/opensim-core/3872#gh-1717684619:-O4NMYFr2ji4qO2bH2k_:b-896fix) ([raw file](https://github.com/opensim-org/opensim-core/blob/418f4d86feff23626383bbe02a9aaa7cd5357290/OpenSim/Simulation/Test/testForces.cpp#L36)):*
<details><summary><i>Previously, tkuchida (Tom Uchida) wrote…</i></summary><blockquote>

9, 10, 10, 11 -> 9, 10, 11, 12
</blockquote></details>

Done.

**Review by AllisonJohn — CHANGES_REQUESTED** [review]
> Reviewed 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3872)* status: all files reviewed, 3 unresolved discussions (waiting on @nickbianco and @tkuchida)

___
*[`OpenSim/Auxiliary/auxiliaryTestFunctions.h` line 280 at r3](https://reviewable.io/reviews/opensim-org/opensim-core/3872#-O4SVdox95iuBhQWwVda:-O4SVdox95iuBhQWwVdb:b9p571q) ([raw file](https://github.com/opensim-org/opensim-core/blob/13b132cb0ca4b64892a1845307f673cf1aec3946/OpenSim/Auxiliary/auxiliaryTestFunctions.h#L280)):*
> ```objc
>                 else{
>                     ap.updValue<string>() = base;
>                 }
> ```

should there be a warning if someone tries to randomize an ExpressionBasedBushingForce?

**Review by nickbianco — COMMENTED** [review]
After fighting with the failing `ExpressionBasedBushingForce` test for a while, I've decided to give up and implement this change without a cache variable now. I've documented the bug here to be fixed in a later PR: https://github.com/opensim-org/opensim-core/issues/3881.

> *[Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3872#-:-O4g6htK8z1zCxBBX6Fp:b-wzfxzw)* status: 6 of 9 files reviewed, 3 unresolved discussions (waiting on @AllisonJohn and @tkuchida)

___
*[`OpenSim/Auxiliary/auxiliaryTestFunctions.h` line 280 at r3](https://reviewable.io/reviews/opensim-org/opensim-core/3872#-O4SVdox95iuBhQWwVda:-O4g5_Um0GK2GpZukKNA:bbesa8h) ([raw file](https://github.com/opensim-org/opensim-core/blob/13b132cb0ca4b64892a1845307f673cf1aec3946/OpenSim/Auxiliary/auxiliaryTestFunctions.h#L280)):*
<details><summary><i>Previously, AllisonJohn wrote…</i></summary><blockquote>

should there be a warning if someone tries to randomize an ExpressionBasedBushingForce?
</blockquote></details>

It's not really being randomized here; rather, it's being assigned an aribitrary string value which is invalid for ExpressionBasedBushingForce. Actually, I'm not really sure why this wasn't failing before. I updated the comment for clarity.

**Review by AllisonJohn — APPROVED** [review]
ok looks good!

> Reviewed 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3872#-:-O4g_fA65DisBD9CZLuO:b-fhtaav)* status: all files reviewed, 2 unresolved discussions (waiting on @tkuchida)

**Review by tkuchida — APPROVED** [review]
LGTM. Some of the `ASSERT_EQUAL`s have strange tolerances but that's unrelated to this PR.

**Review by nickbianco — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3872)* status: all files reviewed, 3 unresolved discussions (waiting on @tkuchida)

___
*[`OpenSim/Simulation/Test/testForces.cpp` line 196 at r4](https://reviewable.io/reviews/opensim-org/opensim-core/3872#gh-1722488015:-O4h3vNb4Zbn7BKbi_YT:b5baklc) ([raw file](https://github.com/opensim-org/opensim-core/blob/8d057ae26c90d920bd157372d0208f069eb5a56b/OpenSim/Simulation/Test/testForces.cpp#L196)):*
<details><summary><i>Previously, tkuchida (Tom Uchida) wrote…</i></summary><blockquote>

This is two orders of magnitude larger than the integrator accuracy (line 167). Was 1e-6 intended (as on line 187)?
</blockquote></details>

Good catch. Updated to 1e-6.

**Comment by nickbianco** [comment]
Thanks @AllisonJohn and @tkuchida!

@mattpetrucci, @stingjp, this is now ready for testing.

**Pull request merged**